src/scheme/WeightedLeastSquares.cpp

```diff
@@ -40,55 +40,86 @@ bool WeightedLeastSquares::solve_(const LinearizedControlProblem & problem,
-                "WeightedLeastSquares does not allow to change the weight of a Task with priority 0.");
-          se.addVectorWeightEvent(c.constraint.get());
-        }
+        fullRebuild = true;
         break;
-        case ProblemDefinitionEvent::Type::TaskAddition:
-          addTask(problem, memory, e.emitter(), se);
+      }
+    }
+    if(fullRebuild)
+    {
+      resetComputationData(problem, memory);
+    }
+    else
+    {
+      while(memory->hasEvents())
+      {
+        auto e = memory->popEvent();
+        switch(e.type())
+        {
+          case EventType::WeightChange: {
+            const auto & c = problem.constraintWithRequirements(e.typedEmitter<EventType::WeightChange>());
+            if(c.requirements->priorityLevel().value() == 0)
+              throw std::runtime_error(
+                  "[WeightedLeastSquares::updateComputationData_] "
+                  "WeightedLeastSquares does not allow to change the weight of a Task with priority 0.");
+            se.addScalarWeightEvent(c.constraint.get());
+          }
           break;
-        case ProblemDefinitionEvent::Type::TaskRemoval:
-          removeTask(problem, memory, e.emitter(), se);
+          case EventType::AnisotropicWeightChange: {
+            const auto & c = problem.constraintWithRequirements(e.typedEmitter<EventType::AnisotropicWeightChange>());
+            if(c.requirements->priorityLevel().value() == 0)
+              throw std::runtime_error(
+                  "[WeightedLeastSquares::updateComputationData_] "
+                  "WeightedLeastSquares does not allow to change the weight of a Task with priority 0.");
+            se.addVectorWeightEvent(c.constraint.get());
+          }
           break;
-        default:
-          throw std::runtime_error("[WeightedLeastSquares::updateComputationData_] Unimplemented event handling.");
+          case EventType::TaskAddition:
+            addTask(problem, memory, e.typedEmitter<EventType::TaskAddition>(), se);
+            break;
+          case EventType::TaskRemoval:
+            removeTask(problem, memory, e.typedEmitter<EventType::TaskRemoval>(), se);
+            break;
+          default:
+            throw std::runtime_error("[WeightedLeastSquares::updateComputationData_] Unimplemented event handling.");
+        }
       }
-    }
 
-    memory->variables(); // update variable vector if needed
-    memory->solver->process(se);
+      memory->variables(); // update variable vector if needed
+      memory->solver->process(se);
+    }
   }
 }
 
 std::unique_ptr<WeightedLeastSquares::Memory> WeightedLeastSquares::createComputationData_(
     const LinearizedControlProblem & problem) const
 {
   auto memory = std::unique_ptr<Memory>(new Memory(id(), solverFactory_->createSolver()));
+  processProblem(problem, memory.get());
+
+  return memory;
+}
+
+void WeightedLeastSquares::resetComputationData(const LinearizedControlProblem & problem, Memory * memory) const
+{
```

Comment:
I think this should also clear the events queue

Reply:
Good catch. Variables also needed to be reset.
Done in 8985dab73f16cfa6fc315be162e9b1dd5e18cdfc